cmake/cc_rules.cmake

```diff
@@ -440,24 +408,88 @@ function(firebase_ios_objc_test target)
   endif()
 endfunction()
 
-# firebase_ios_generate_dummy_source(name, sources_list)
+# firebase_ios_generate_dummy_source(target, sources_list)
 #
 # Generates a dummy source file containing a single symbol, suitable for use as
 # a source file in when defining a header-only library.
 #
-# Appends the generated source file name to the list named by sources_list.
-macro(firebase_ios_generate_dummy_source name sources_list)
-  set(__empty_header_only_file "${CMAKE_CURRENT_BINARY_DIR}/${name}_header_only_empty.cc")
+# Appends the generated source file target to the list named by sources_list.
+macro(firebase_ios_generate_dummy_source target sources_list)
+  set(
+    __empty_header_only_file
+    "${CMAKE_CURRENT_BINARY_DIR}/${target}_header_only_empty.cc"
+  )
 
   if(NOT EXISTS ${__empty_header_only_file})
     file(WRITE ${__empty_header_only_file}
       "// Generated file that keeps header-only CMake libraries happy.
 
       // single meaningless symbol
-      void ${name}_header_only_fakesym() {}
+      void ${target}_header_only_fakesym() {}
       "
     )
   endif()
 
   list(APPEND ${sources_list} ${__empty_header_only_file})
 endmacro()
+
+# Splits the given arguments into two lists: those suitable for passing to
+# `firebase_ios_set_common_target_options` and everything else, usually passed
+# to `add_library`, `add_executable`, or similar.
+#
+# The resulting lists are set in the parent scope as `${prefix}_OPTIONS` for use
+# when calling `firebase_ios_set_common_target_options`, and `${prefix}_REST`
+# that contains any arguments that were unrecognized.
+#
+# See `firebase_ios_set_common_target_options` for details on which arguments
+# are split out.
+function(firebase_ios_split_target_options prefix)
+  set(option_names DISABLE_STRICT_WARNINGS)
+
+  set(${prefix}_OPTIONS)
+  set(${prefix}_REST)
+
+  foreach(arg ${ARGN})
+    list(FIND option_names ${arg} option_index)
+    if(NOT option_index EQUAL -1)
+      list(APPEND ${prefix}_OPTIONS ${arg})
+    else()
+      list(APPEND ${prefix}_REST ${arg})
+    endif()
+  endforeach()
+
+  set(${prefix}_OPTIONS ${${prefix}_OPTIONS} PARENT_SCOPE)
+  set(${prefix}_REST ${${prefix}_REST} PARENT_SCOPE)
+endfunction()
+
+# Sets common options for a given target. This includes:
+#
+#   * `DISABLE_STRICT_WARNINGS`: disables stricter warnings usually applied to
+#     targets in this repo. Use this to compile code that's local but doesn't
+#     hold itself to Firestore's warning strictness.
+#
+# All other arguments are ignored.
+function(firebase_ios_set_common_target_options target)
+  set(bool DISABLE_STRICT_WARNINGS)
+  cmake_parse_arguments(flag "${bool}" "" "" ${ARGN})
+
+  set(suffix _STRICT)
```

Comment:
To improve readability, consider calling `set(suffix _STRICT)` in an `else` block instead of defining it before the `if` block, then potentially overwriting it.